### DISCUSSION TITLE
maillog_maillog is getting large/slow/eats cpu+ram

### DISCUSSION THREAD

**Discussion by harrykaine** [modoboa/modoboa#2424] [Help] [answered]
Hi,
My dedicated mail server stops working due to  the load generated by postgres. 
After a reload of pg (and the connected services) everything is fine, but load starts slowly to rise.
I"ve installed pg_stat_statements and realized that accesses to maillog_maillog  seem to be the  source.
`
SELECT "maillog_maillog"."id", "maillog_maillog"."queue_id", "maillog_maillog"."date", "maillog_maillog"."sender", "maillog_maillog"."rcpt", "maillog_maillog"."original_rcpt", "maillog_maillog"."size", "maillog_maillog"."status", "maillog_maillog"."from_domain_id", "maillog_maillog"."to_domain_id" FROM "maillog_maillog" ORDER BY "maillog_maillog"."date" DESC  LIMIT .....
`
It seems to be, that all mails since the start are registered in this table and its size is growing. 

`
select count(*) from maillog_maillog  ;
  count
---------
 1169416
(1 row)
`
Question: Can I delete some of the older entries or will something bad happen?

Thanks

**Comment by tonioo** [comment, marked as answer]
@harrykaine Juste create an index on the date field in the maillog_maillog table and it should work better

**Reply by harrykaine** [reply]
Thank you very much.... that worked!
